frontend-vue/src/stores/auth.ts

```diff
@@ -0,0 +1,149 @@
+import { Log, UserManager, UserManagerSettings, WebStorageStateStore } from "oidc-client-ts";
+import { defineStore } from "pinia";
+import { Ref, computed, inject, onMounted, ref } from "vue";
+import { AfterFetchContext, createFetch } from "@vueuse/core";
+
+export interface UserInfo {
+  authenticated: boolean;
+  name?: string;
+  permissions: string[];
+}
+
+interface OidcConfiguration {
+  securityType: "oidc";
+  authority: string;
+  clientId: string;
+  clientSecret: string;
+}
+
+interface SecuirtyDisabled {
+  securityType: "disabled";
+}
+
+type SecurityConfiguration = OidcConfiguration | SecuirtyDisabled;
+
+const unauthenticatedUser: UserInfo = {
+  authenticated: false,
+  permissions: [],
+};
+
+const useFetch = createFetch({
+  baseUrl: import.meta.env.VITE_CONTEXTURE_API_BASE_URL,
+  fetchOptions: {
+    redirect: "follow",
+  },
+
+  options: {
+    async afterFetch(ctx: AfterFetchContext) {
+      return {
+        response: ctx.response,
+        data: typeof ctx.data === "object" ? ctx.data : JSON.parse(ctx.data),
+      };
+    },
+    async onFetchError(ctx: { data: any; response: Response | null; error: any }) {
+      console.warn(ctx);
+      return ctx;
+    },
+  },
+});
+
+export async function getSecurityConfiguration() {
+  const { data } = await useFetch<SecurityConfiguration>("/meta/securityConfiguration").get();
```

Comment:
Currently it's a bit problematic to use any custom configuration with FE when running contexture from docker image since its a build time dependency.
I plan on revisiting the auth and login support in the near future I'd like to investigate the idea of supporting scenario where auth proxy is used for login and token management - this could allow implementing custom authentication/authorization requirements externally without involving changes to Contexture.API